packages/tailwind/tailwind-base.css

```
@@ -196,6 +201,64 @@
 }
 
 @layer components {
+  /*** Grid Layout ***/
+
+  .layout-grid,
+  .layout-grid-container {
+    /** Grid base, defines the grid columns and their spacing */
+    @apply layout-grid-gap-x sm:grid-cols-14 grid;
+
+    /** Grid container that applies the grid columns, grid spacing and the container component class */
+    &.layout-grid-container {
+      @apply container;
+      /** Make content of the grid span the mid 12 columns as a default (unless it has another explicit col-span class) */
+      & > * {
+        @apply sm:col-start-2 sm:col-span-12;
+      }
+    }
+  }
+
+  [class^="layout-subgrid-"] {
+    @apply layout-grid-gap-x grid;
+  }
+
+  .layout-subgrid-1 {
```

kunne dette vært en utility? 



```suggestion
@utility layout-subgrid-* {
	@apply sm:grid-cols-*
```

mulig ikke det funker med apply i en utility, men kanskje grid-cols-[--value(integer)] funker 🤔

---

ref https://tailwindcss.com/docs/adding-custom-styles#bare-values

---

Ikke sant. Jeg startet med `utility` `--value(integer)`, men så ja: da kom jeg på at vi kanskje ikke vil støtte mer enn opp til 12. Det vil funke opp til 14 for så vidt, men etter det vil det jo bli helt fucked hvis man har en subgrid som plutselig er 16 kolonner inne i en med 14.

Men jeg lurer også på om det kanskje hadde vært smart å droppe `sm:` her, slik at consumenten kan styre breakpoints? Det er mange ganger jeg selv setter en høyere breakpoint inne i subgrids, fordi det ofte passer bedre jo dypere nestet ting er

---

aha aight. ja, jeg er kanskje enig i det. Kjøre uten sm så får folk heller passe på selv?

---

Ja, veldig enig der! Vi dropper `sm`